ethers/contract.nim

```diff
@@ -116,14 +117,46 @@ proc call(contract: Contract,
   let response = await contract.provider.call(transaction, overrides)
   return decodeResponse(ReturnType, returnMultiple, response)
 
+proc populateTransaction(
```

Comment:
I would expect the logic to handle stuck transactions to be at the level of the signer, not the level of the contract invocations? Stuck transactions can happen also when using the signer directly, I think?

Reply:
I thought about this too, but ultimately I feel that almost every transaction that may run into this issue originates from a contract transactions, and any others would have to be initiated from a user creating a transaction and submitting it through the signer, in which case they would get the `EstimateGasError` and they can handle the error as they see fit, including calling `cancelTransction` (which is on the level of the signer).

Reply:
This logic has been removed.